### PR TITLE
Document the field flag for create-setter

### PR DESCRIPTION
**Pull request by mortent**

### CODE REVIEW

File: site/content/en/guides/producer/setters/_index.md [hunk 1]
```diff
@@ -87,6 +87,51 @@ spec:
...
+    replicas: 3
+spec:
+  replicas: 3 # {"$kpt-set":"replicas"}
+```
+
+The path is always just the names of the properties on the path to the field,
+regardless whether the field is nested inside a sequence.
+
+```sh
+# create a setter named "mountName" and only target the name of the volume mount
```
Comment (phanimarupaka) [inline site/content/en/guides/producer/setters/_index.md:116]: Probably modify this example to have multiple list values for volumeMounts and describe the current behavior. We might want to have action items to make it the syntax more expressive.

File: docs/guides/producer/index.xml [hunk 2]
```diff
@@ -155,6 +155,36 @@ kpt cfg create-setter hello-world/ replicas &lt;span style=&#34;color:#0000cf;fo
 &lt;/span&gt;&lt;span style=&#34;color:#f8f8f8;text-decoration:underline&#34;&gt;  &lt;/span&gt;&lt;span style=&#34;color:#204a87;font-weight:bold&#34;&gt;name&lt;/span&gt;&lt;span style=&#34;color:#000;font-weight:bold&#34;&gt;:&lt;/span&gt;&lt;span style=&#34;color:#f8f8f8;text-decoration:underline&#34;&gt; &lt;/span&gt;foo&lt;span style=&#34;color:#f8f8f8;text-decoration:underline&#34;&gt;
```
Comment (frankfarzan) [inline docs/guides/producer/index.xml:155]: Long overdue: https://github.com/GoogleContainerTools/kpt/issues/940

File: site/content/en/guides/producer/setters/_index.md [hunk 3]
```diff
@@ -87,6 +87,51 @@ spec:
   replicas: 3 # {"$kpt-set":"replicas"}
 ```
 
+#### Targeting fields using the path
+
+To limit the fields targeted by the setter, use the `--field` flag to provide
```
Comment (frankfarzan) [inline site/content/en/guides/producer/setters/_index.md:92]: Describe the interaction with selecting based on value.

File: site/content/en/guides/producer/setters/_index.md [hunk 4]
```diff
@@ -87,6 +87,51 @@ spec:
...
+
+```yaml
+# deployment-nginx.yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: nginx-deployment
+spec:
+  containers:
+  - name: nginx
```
Comment (frankfarzan) [inline site/content/en/guides/producer/setters/_index.md:128]: Interesting part is to show another element in the array here and whether its property is selected.

File: site/content/en/guides/producer/setters/_index.md [hunk 5]
```diff
@@ -87,6 +87,51 @@ spec:
...
+spec:
+  replicas: 3 # {"$kpt-set":"replicas"}
+```
+
+The path is always just the names of the properties on the path to the field,
+regardless whether the field is nested inside a sequence.
+
+```sh
+# create a setter named "mountName" and only target the name of the volume mount
+kpt cfg create-setter hello-world/ mountName nginx --field="spec.containers.volumeMounts.name"
```
Comment (frankfarzan) [inline site/content/en/guides/producer/setters/_index.md:117]: Correct me if wrong, but path expression here is not the same used in 'kpt cfg grep'?

Search/Replace proposal will remove this discrepancy.

File: site/content/en/guides/producer/setters/_index.md (hunk 4 above)
Comment (phanimarupaka) [reply site/content/en/guides/producer/setters/_index.md:128]: If both field path and value match, then they will be selected.

File: site/content/en/guides/producer/setters/_index.md (hunk 5 above)
Comment (phanimarupaka) [reply site/content/en/guides/producer/setters/_index.md:117]: They are supposed to be same but I see an error if I grep. We should correct it and make the behavior consistent.

File: site/content/en/guides/producer/setters/_index.md (hunk 4 above)
Comment (mortent) [reply site/content/en/guides/producer/setters/_index.md:132]: So in this example, without using the `--field` flag, several other fields would have been selected.

File: site/content/en/guides/producer/setters/_index.md (hunk 1 above)
Comment (mortent) [reply site/content/en/guides/producer/setters/_index.md:120]: Added an additional volumeMount to the example. Added more explanation further up on how targeting works without the `--field` flag.

**Comment by phanimarupaka** [comment]
looks good to me

File: site/content/en/guides/producer/setters/_index.md [hunk 6]
```diff
@@ -87,6 +87,57 @@ spec:
...
 ```
 
+#### Targeting fields using the path
+
+The basic way to create a setter only matches fields based on the value. But
+in some situations there might be several fields with the same value, but not all
+of them should be targeted by the setter. In these situations, use the `--field`
+flag to provide either the name of the field, the full path to the field, or a
+partial (suffix) path to the field. Only fields that match the path will be
+targeted by the setter.
```
Comment (frankfarzan) [inline site/content/en/guides/producer/setters/_index.md:97]: I think what's implicit and must be said upfront is the AND logic here. Both Value AND path must match. In the example below also highlight this.

Comment (mortent) [reply site/content/en/guides/producer/setters/_index.md]: Updated.

**Review by phanimarupaka — APPROVED** [review]

**Review by frankfarzan — APPROVED** [review]

**Pull request merged**